api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -69,10 +70,20 @@
   }
 
   @Override
-  public <C> void inject(SpanContext spanContext, C carrier, Setter<C> setter) {
-    checkNotNull(spanContext, "spanContext");
+  public <C> void inject(Context context, C carrier, Setter<C> setter) {
```

Comment:
Looks like this inject(Context, ...) and extract(Context, ...) could be moved to some abstract base class like `SingleAspectPropagator<SpanContext>`.

Reply:
We can do that once we have the `Binary` format back in place. Right now it's kinda useless IHMO (and we *do have* to add the `Binary` format back).